Finger Curl Calculation Addition

chetu3319 opened the pull request:
Adding functions to calculate the curl of fingers. 
Example usage: 
```
// Compute the curl of Index Finger of Right Hand
float indexCurl =  HandPoseUtils.IndexCurl(Handedness.Right);
```

Output ranges from 0 (No-Curl) to 1 (Fully-Curled) finger. 

<img src="https://media.giphy.com/media/gHEOfqYUMR19OfyMWw/giphy-downsized-large.gif" width = "600" height = 600 />

chetu3319 committed:
Merge pull request #1 from provencher/master

Rebase

Commit:
Finger Curl Calculation

Addition of functions to calculate the curl of fingers in HandPoseUtils script.
The output value provides curl value of a finger ranging 0 to 1, where 1 is finger curled.

Commit:
Added Credtis to HandPoseUtils

provencher commented on Assets/MRTK.ThirdParty/MRTK-Quest/Scripts/Debug/FingerCurlExample.cs:
```diff
@@ -0,0 +1,37 @@
+    public GameObject ringFinger;
+    public GameObject pinkyFinger;
+
+
```
please remove some of the white space here.

provencher commented on Assets/MRTK.ThirdParty/MRTK-Quest/Scripts/Utils/HandPoseUtils.cs:
```diff
@@ -90,5 +94,98 @@ public static bool IsThumbGrabbing(Handedness hand)
+        * Author: Chaitanya Shah
+        * github: https://github.com/chetu3319
+        */
+
```
Please remove one white space here. Try not to keep more than one blank line before spaced out areas here.

provencher commented on Assets/MRTK.ThirdParty/MRTK-Quest/Scripts/Debug/FingerCurlExample.cs:
```diff
@@ -0,0 +1,37 @@
+{
+    public Handedness handedness;
+
+    public GameObject indexFinger;
```
My preference is to have private serialized fields.

provencher commented on Assets/MRTK.ThirdParty/MRTK-Quest/Scripts/Debug/FingerCurlExample.cs:
```diff
@@ -0,0 +1,37 @@
+using prvncher.MixedReality.Toolkit.Utils;
+using Microsoft.MixedReality.Toolkit.Utilities;
+
+public class FingerCurlExample : MonoBehaviour
```
Please add a namespace for this class. See the other debug sample classes

provencher commented on Assets/MRTK.ThirdParty/MRTK-Quest/Scripts/Debug/FingerCurlExample.cs:
```diff
@@ -0,0 +1,37 @@
+﻿using System.Collections;
```
Please add a header here spelling out the license and such. See other classes in this project for a reference.

provencher reviewed (changes requested):
Code looks great! Some minor changes.
If you can post a gif or a screenshot of what the scene changes look like, it would be great.

provencher commented on Assets/MRTK.ThirdParty/MRTK-Quest/Scripts/Utils/HandPoseUtils.cs:
```diff
@@ -90,5 +94,98 @@ public static bool IsThumbGrabbing(Handedness hand)
+            var palmToFinger = (fingerKnuckleJoint - wristJoint).normalized;
+            var fingerKnuckleToTip = (fingerKnuckleJoint - fingerTipJoint).normalized;
+
+            var curl = Vector3.Dot(fingerKnuckleToTip, palmToFinger);
```
Clever approach to do a dot product here!

Commit:
Clean up

Code clean up.
- Adding license info
- Cleaning up white spaces
- Reformatting according to best coding practices.

Commit:
Grip Visual Update

- Updating hand material visual cue to support float grip strength.
- Refactoring function names making it self explainatory

chetu3319 commented:
I've made the suggested updates of calculating `gripStrength` from the finger's curl value. 
I carried two different calculation: 
1. Only considering the index and middle finger's curl value. Works for full hand grip, but provides a false visual cue when only that two fingers are curled. 
2. Considering all four fingers' curl value. Works perfectly providing intended visual cue.
